### PR TITLE
#1543  Use interpolation expressions instead of the v-html attribute

### LINKED ISSUE #1543: (JS-0693) Use interpolation expressions instead of the `v-html` attribute
## Description
It is recommended to use interpolation expressions instead of using `v-html` as it prevents injection attacks like XSS.

## Occurrences
There are 5 occurrences of this issue in the repository.

See all occurrences on DeepSource &rarr; [deepsource.io/gh/kodadot/nft-gallery/issue/JS-0693/occurrences/](https://deepsource.io/gh/kodadot/nft-gallery/issue/JS-0693/occurrences/)

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇  _ Do a quick check before the merge. 

### PR type
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### Before submitting Pull Request, please make sure:
- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything works
- [ ] I found edge cases

### What's new?
- [x] PR closes #1543
- [ ] 
Removed one useless FAQ file that was not being used anywhere, it also had a v-html tag
the real FAQ file still has one, since that requires parsing the text, I don't think v-html would be an issue here, but if we were to remove it, we will have to parse it through `domParser`, let me know if we want to go this way.
Removed it from series-insights

### Had issue bounty label ?
- [x] Fill up your KSM address: 
[Payout](https://kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

### Community participation
- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
- [ ] My fix has changed **something** on UI, a screenshot for others, is best to understand changes.

### CODE REVIEW

File: components/series/SeriesTable.vue [hunk 1]
```diff
@@ -134,13 +134,15 @@
       >
         <template v-if="!isLoading">
           <div
-            v-html="
+            :class="getClassForVolumne()"
```
Comment (kkukelka) [inline components/series/SeriesTable.vue:137]:
```suggestion
            :class="getClassForVolume()"
```
spelling error 😛

File: components/series/SeriesTable.vue [hunk 2]
```diff
@@ -155,14 +157,16 @@
         :visible="nbDays === '7'"
       >
         <template v-if="!isLoading">
-          <div
-            v-html="
+         <div
+            :class="getClassForVolumne()"
```
Comment (kkukelka) [inline components/series/SeriesTable.vue:161]:
```suggestion
            :class="getClassForVolume()"
```

File: components/series/SeriesTable.vue [hunk 3]
```diff
@@ -177,14 +181,16 @@
         :visible="nbDays === '30'"
       >
         <template v-if="!isLoading">
-          <div
-            v-html="
+         <div
+            :class="getClassForVolumne()"
```
Comment (kkukelka) [inline components/series/SeriesTable.vue:185]:
```suggestion
            :class="getClassForVolume()"
```

File: components/series/SeriesTable.vue [hunk 4]
```diff
@@ -406,10 +413,14 @@ export default class SeriesTable extends Vue {
...
-    return volumePercent > 0
-      ? `<div style="color: #41b883"> +${volumePercent}%</div>`
-      : `<div class="has-text-danger"> ${volumePercent}%</div>`
+    this.volumePercent = Math.round(vol * 100) / 100
+    return this.volumePercent > 0
+      ? `+${this.volumePercent}%`
+      : `${this.volumePercent}%`
+  }
+
+  public getClassForVolumne() {
```
Comment (kkukelka) [inline components/series/SeriesTable.vue:422]:
```suggestion
  public getClassForVolume() {
```

**Review by kkukelka — CHANGES_REQUESTED** [review]
just some spelling errors

**Review by kkukelka — APPROVED** [review]

File: components/series/SeriesTable.vue [hunk 5]
```diff
@@ -421,4 +432,8 @@ export default class SeriesTable extends Vue {
   color: #000;
   background-color: $primary;
 }
+
+.has-text-increase {
```
Comment (vikiival) [inline components/series/SeriesTable.vue:436]: may use `has-text-success` from bulma?

File: components/series/SeriesTable.vue [hunk 6]
```diff
@@ -406,10 +413,14 @@ export default class SeriesTable extends Vue {
...
-      : `<div class="has-text-danger"> ${volumePercent}%</div>`
+    this.volumePercent = Math.round(vol * 100) / 100
+    return this.volumePercent > 0
+      ? `+${this.volumePercent}%`
+      : `${this.volumePercent}%`
+  }
+
+  public getClassForVolume() {
+    return this.volumePercent > 0 ? 'has-text-increase' : 'has-text-danger'
   }
```
Comment (roiLeo) [inline components/series/SeriesTable.vue:424]: What will happen if volumePercent === 0 or is undefined? we will be seing '---' with red text?

Comment (roiLeo) [reply components/series/SeriesTable.vue:424]: I'm having issue while loading series-insight page

Comment (prachi00) [reply components/series/SeriesTable.vue:424]: yeah me too, on local, so wasn't able to test it

Comment (prachi00) [reply components/series/SeriesTable.vue:424]: and yes it will return '---'

Comment (yangwao) [reply components/series/SeriesTable.vue]: we have some issues with a subquery (indexer) and once we deploy subsquid indexer, we should get back these big queries, hold on patience :D we've crossed their limits with size of our payload, well happens.

**Comment by yangwao** [comment]
is it good to go? 
Why was FAQ deleted 👀

**Comment by prachi00** [comment]
> is it good to go? Why was FAQ deleted 👀

@yangwao it was a repetition, the original file is in a different folder, this one is not getting used anywhere

**Comment by roiLeo** [comment]
Hey, '---' with red text isn't fixed yeet, see my review.
Screenshot 2021-12-26 at 09-27-59 nft-gallery-nuxt

**Comment by prachi00** [comment]
> Hey, '---' with red text isn't fixed yeet, see my review. Screenshot 2021-12-26 at 09-27-59 nft-gallery-nuxt

@roiLeo Have fixed this

**Comment by roiLeo** [comment]
> @roiLeo Have fixed this

Screenshot 2021-12-26 at 11-43-08 nft-gallery-nuxt

For test case, change `collectionSeriesList.graphql`
```
  collectionEntities(
    first: 9
    offset: 0
    filter: { issuer: { notIn: $denyList } }
  ) {
```

**Comment by prachi00** [comment]
> > @roiLeo Have fixed this
> 
> Screenshot 2021-12-26 at 11-43-08 nft-gallery-nuxt
> 
> For test case, change `collectionSeriesList.graphql`
> 
> ```
>   collectionEntities(
>     first: 9
>     offset: 0
>     filter: { issuer: { notIn: $denyList } }
>   ) {
> ```

@roiLeo How are you opening the 

> > @roiLeo Have fixed this
> 
> Screenshot 2021-12-26 at 11-43-08 nft-gallery-nuxt
> 
> For test case, change `collectionSeriesList.graphql`
> 
> ```
>   collectionEntities(
>     first: 9
>     offset: 0
>     filter: { issuer: { notIn: $denyList } }
>   ) {
> ```

@roiLeo How did you open the series-insight page? I am not able to test it as the route doesnt open for me

**Comment by prachi00** [comment]
> first: 9
>     offset: 0
>     filter: { issuer: { notIn: $denyList } }



> > > @roiLeo Have fixed this
> > 
> > 
> > Screenshot 2021-12-26 at 11-43-08 nft-gallery-nuxt
> > For test case, change `collectionSeriesList.graphql`
> > ```
> >   collectionEntities(
> >     first: 9
> >     offset: 0
> >     filter: { issuer: { notIn: $denyList } }
> >   ) {
> > ```
>

**Comment by prachi00** [comment]
@roiLeo The logic for this seems fine to me, can you also once take a look and see if something seems off? Not able to figure out why its working incorrectly

**Comment by prachi00** [comment]
> @roiLeo The logic for this seems fine to me, can you also once take a look and see if something seems off? Not able to figure out why its working incorrectly

@roiLeo changed the logic a little bit, seems to work fine now

**Review by roiLeo — APPROVED** [review]
Well, it seems to work. I find the logic difficult to understand but we can improve it later.

Note: I found bugs with the numbers displayed (empty buys columns, floor price sorting...)

**Comment by prachi00** [comment]
> Well, it seems to work. I find the logic difficult to understand but we can improve it later.

Added comments to make it easier to understand

**Comment by yangwao** [comment]
Perfect, I’ve sent the payout https://kusama.subscan.io/extrinsic/0x557ab67e55aafb5bd76d2fa53145540d576313eebae6727f95d6e29ba66ec2ba

Let’s grab another issue and get rewarded! 
https://github.com/kodadot/nft-gallery/issues/

**Pull request merged**